patches/react-native-share-menu+6.0.0.patch

```diff
@@ -1,13 +1,94 @@
+@@ -112,6 +112,14 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
+     clearSharedText();
+   }
+ 
++  @ReactMethod
++  public void addListener(String eventName) {
++  }
++
++  @ReactMethod
++  public void removeListeners(Integer count) {
++  }
++
+   private void dispatchEvent(ReadableMap shared) {
+     if (mReactContext == null || !mReactContext.hasActiveCatalystInstance()) {
+       return;
+diff --git a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+index e290cce..44e5dfe 100644
+--- a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
++++ b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+@@ -34,6 +34,8 @@ public class ShareMenuReactView: NSObject {
+             return
+         }
+ 
++        ShareMenuReactView.detachViewDelegate()
++
+         if error != nil {
+             let exception = NSError(
+                 domain: Bundle.main.bundleIdentifier!,
 diff --git a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
-index f42bce6..ee36062 100644
+index f42bce6..ab00813 100644
 --- a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
 +++ b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
 @@ -13,7 +13,7 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
    func sourceURL(for bridge: RCTBridge!) -> URL! {
  #if DEBUG
      return RCTBundleURLProvider.sharedSettings()?
 -      .jsBundleURL(forBundleRoot: "index.share", fallbackResource: nil)
-+      .jsBundleURL(forBundleRoot: "index.share")
++      .jsBundleURL(forBundleRoot: "index")
  #else
      return Bundle.main.url(forResource: "main", withExtension: "jsbundle")
  #endif
+@@ -29,20 +29,7 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
+       initialProperties: nil
+     )
+ 
+-    rootView.backgroundColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
+-    backgroundColorSetup: if let backgroundColorConfig = Bundle.main.infoDictionary?[REACT_SHARE_VIEW_BACKGROUND_COLOR_KEY] as? [String:Any] {
+-      if let transparent = backgroundColorConfig[COLOR_TRANSPARENT_KEY] as? Bool, transparent {
+-        rootView.backgroundColor = nil
+-        break backgroundColorSetup
+-      }
+-
+-      let red = backgroundColorConfig[COLOR_RED_KEY] as? Float ?? 1
+-      let green = backgroundColorConfig[COLOR_GREEN_KEY] as? Float ?? 1
+-      let blue = backgroundColorConfig[COLOR_BLUE_KEY] as? Float ?? 1
+-      let alpha = backgroundColorConfig[COLOR_ALPHA_KEY] as? Float ?? 1
+-
+-      rootView.backgroundColor = UIColor(red: CGFloat(red), green: CGFloat(green), blue: CGFloat(blue), alpha: CGFloat(alpha))
+-    }
++    RCTBootSplash.initWithStoryboard("MainInterface", rootView: rootView)
```

Comment:
Ah, is this why the changes to the storyboard file? Should that be part of the patch so everyone using the library would get this? I honestly don't know anything about storyboard. What's the motivation for this change?

Reply:
Mostly just trying to replicate as much of the setup from AppDelegate as we can. In release mode there's no pause waiting for metro so you just get a very brief flash of this before you're already in the app. We might be able to remove it just need to make sure the calls to hide the bootsplash later don't blow up.